TypeAlias defined inside functions doesn't respect the typealias naming style.

### Bug description

Example code:

```python
# pylint: disable=missing-docstring
from typing import TypeAlias

GoodTopLevelName: TypeAlias = int
bad_top_level_name: TypeAlias = int


def my_func():
    GoodLocalName: TypeAlias = int
    bad_local_name : TypeAlias = int
    del GoodLocalName, bad_local_name
```

### Configuration

_No response_

### Command used

```shell
pylint t.py
```


### Pylint output

```shell
t.py:5:0: C0103: Type alias name "bad_top_level_name" doesn't conform to predefined naming style (invalid-name)
```


### Expected behavior

It should also raise `invalid-name` for the `bad_local_name ` variable

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.2
Python 3.11.2 (main, Mar  3 2023, 16:02:46) [GCC 12.2.0]
```


### OS / Environment

_No response_

### Additional dependencies

I having a PR coming.

--- a/doc/whatsnew/fragments/8536.false_negative
+++ b/doc/whatsnew/fragments/8536.false_negative
@@ -1,0 +1,3 @@
+`TypeAlias` variables defined in functions are now checked for `invalid-name` errors.
+
+Closes #8536

--- a/pylint/checkers/base/name_checker/checker.py
+++ b/pylint/checkers/base/name_checker/checker.py
@@ -476,7 +476,12 @@ class NameChecker(_BasicChecker):
             # global introduced variable aren't in the function locals
             if node.name in frame and node.name not in frame.argnames():
                 if not _redefines_import(node):
-                    self._check_name("variable", node.name, node)
+                    if isinstance(
+                        assign_type, nodes.AnnAssign
+                    ) and self._assigns_typealias(assign_type.annotation):
+                        self._check_name("typealias", node.name, node)
+                    else:
+                        self._check_name("variable", node.name, node)
 
         # Check names defined in class scopes
         elif isinstance(frame, nodes.ClassDef):

--- a/tests/functional/t/typealias_naming_style_default.py
+++ b/tests/functional/t/typealias_naming_style_default.py
@@ -31,3 +31,14 @@ x: Union[str, int] = 42
 y: Union[str, int]
 # But the following, using a good TypeAlias name, is:
 GoodTypeAliasToUnion: TypeAlias = Union[str, int]
+
+
+def my_function():
+    """My doc."""
+    LocalGoodName: TypeAlias = int
+    local_bad_name: TypeAlias = int  # [invalid-name]
+    local_declaration: Union[str, int]
+    LocalTypeAliasToUnion: TypeAlias = Union[str, int]
+    local_declaration = 1
+    del local_declaration
+    del LocalGoodName, local_bad_name, LocalTypeAliasToUnion

--- a/tests/functional/t/typealias_naming_style_default.txt
+++ b/tests/functional/t/typealias_naming_style_default.txt
@@ -9,4 +9,4 @@ invalid-name:23:0:23:9::"Type alias name ""_BAD_NAME"" doesn't conform to predef
 invalid-name:24:0:24:10::"Type alias name ""__BAD_NAME"" doesn't conform to predefined naming style":HIGH
 invalid-name:25:0:25:9::"Type alias name ""_1BadName"" doesn't conform to predefined naming style":HIGH
 invalid-name:26:0:26:14::"Type alias name ""ANOTHERBADNAME"" doesn't conform to predefined naming style":HIGH
-invalid-name:36:4:36:18:my_function:"Type alias name ""local_bad_name"" doesn't conform to predefined naming style":HIGH
+invalid-name:39:4:39:18:my_function:"Type alias name ""local_bad_name"" doesn't conform to predefined naming style":HIGH

--- a/tests/functional/t/typealias_naming_style_default.txt
+++ b/tests/functional/t/typealias_naming_style_default.txt
@@ -9,3 +9,4 @@ invalid-name:23:0:23:9::"Type alias name ""_BAD_NAME"" doesn't conform to predef
 invalid-name:24:0:24:10::"Type alias name ""__BAD_NAME"" doesn't conform to predefined naming style":HIGH
 invalid-name:25:0:25:9::"Type alias name ""_1BadName"" doesn't conform to predefined naming style":HIGH
 invalid-name:26:0:26:14::"Type alias name ""ANOTHERBADNAME"" doesn't conform to predefined naming style":HIGH
+invalid-name:36:4:36:18:my_function:"Type alias name ""local_bad_name"" doesn't conform to predefined naming style":HIGH

